### DISCUSSION TITLE
How to (artificially) provoke drift in a given dataset?

### DISCUSSION THREAD

**Discussion by TxusLopez** [online-ml/river#569] [Q&A]
Hi,

Is there any way of (artificially) provoking drift in any given dataset?

Thanks in advance

**Comment by MaxHalford** [comment]
There's `ConceptDriftStream` which you can find [here](https://github.com/online-ml/river/blob/master/river/datasets/synth/concept_drift_stream.py) (not sure why it's not showing in the docs). We still have some effort to put in this direction :).

The [stream-learn](https://github.com/w4k2/stream-learn) package does this quite well. We proposed to them to merge with us, but they sort of refused.

**Reply by TxusLopez** [reply]
Hi,

thanks for the prompt reply. 

Yes, I know that function, but it seems like it was to join two or more synthetic datasets created with river library (river.datasets.base.SyntheticDataset). 

I was wondering whether I could add one or several drifts to, e.g. my own real-world dataset. It is difficult to obtain real datasets with known drifts (detected), so I need to artificially provoke it, so that I can have my learning algorithms ready to perform well  when a real drift occurs.

Thanks!
